tests/components/climate/test_dyson.py

```diff
@@ -0,0 +1,371 @@
+    device.state.focus_mode = FocusMode.FOCUS_OFF.value
+    return device
+
+
+def _get_device_cool():
+    """Return a device with state of cooling."""
+    device = mock.Mock()
+    device.name = "Device_name"
+    device.state.tilt = TiltState.TILT_FALSE.value
+    device.state.focus_mode = FocusMode.FOCUS_OFF.value
+    device.state.heat_target = HeatTarget.celsius(12)
+    device.state.heat_mode = HeatMode.HEAT_OFF.value
+    device.state.heat_state = HeatState.HEAT_STATE_OFF.value
+    device.environmental_state.temperature = 288
+    device.environmental_state.humidity = 53
+    return device
+
+
+def _get_device_heat_off():
+    """Return a device with state of heat reached target."""
+    device = mock.Mock()
+    device.name = "Device_name"
+    device.state = mock.Mock()
+    device.state.tilt = TiltState.TILT_FALSE.value
+    device.state.focus_mode = FocusMode.FOCUS_ON.value
+    device.state.heat_target = HeatTarget.celsius(20)
+    device.state.heat_mode = HeatMode.HEAT_ON.value
+    device.state.heat_state = HeatState.HEAT_STATE_OFF.value
+    device.environmental_state.temperature = 293
+    device.environmental_state.humidity = 53
+    return device
+
+
+def _get_device_heat_on():
+    """Return a device with state of heating."""
+    device = mock.Mock(spec=DysonPureHotCoolLink)
+    device.name = "Device_name"
+    device.state = mock.Mock()
+    device.state.tilt = TiltState.TILT_FALSE.value
+    device.state.focus_mode = FocusMode.FOCUS_ON.value
+    device.state.heat_target = HeatTarget.celsius(23)
+    device.state.heat_mode = HeatMode.HEAT_ON.value
+    device.state.heat_state = HeatState.HEAT_STATE_ON.value
+    device.environmental_state.temperature = 289
+    device.environmental_state.humidity = 53
+    return device
+
+
+class DysonTest(unittest.TestCase):
+    """Dyson Climate component test class."""
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Setup things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    @mock.patch('libpurecoollink.dyson.DysonAccount.devices',
+                return_value=[_get_device_heat_on(), _get_device_cool()])
+    @mock.patch('libpurecoollink.dyson.DysonAccount.login', return_value=True)
+    def test_setup_component_with_parent_discovery(self, mocked_login,
+                                                   mocked_devices):
+        """Test setup_component using discovery."""
+        # dyson.setup(self.hass, )
+        setup_component(self.hass, dyson_parent.DOMAIN, {
+            dyson_parent.DOMAIN: {
+                dyson_parent.CONF_USERNAME: "email",
+                dyson_parent.CONF_PASSWORD: "password",
+                dyson_parent.CONF_LANGUAGE: "US",
+                }
+            })
+        self.assertEqual(len(self.hass.data[dyson.DYSON_DEVICES]), 2)
+
```

Comment:
Can we assert here that `device._device.add_message_listener` was called?

Reply:
Yes!